domain/miningmanager/mempool/orphan_pool.go

```diff
@@ -329,17 +333,58 @@ func (op *orphansPool) randomNonHighPriorityOrphan() *model.OrphanTransaction {
 	return nil
 }
 
-func (op *orphansPool) getOrphanTransaction(transactionID *externalapi.DomainTransactionID) (*externalapi.DomainTransaction, bool) {
+func (op *orphansPool) getOrphanTransaction(transactionID *externalapi.DomainTransactionID, clone bool) (*externalapi.DomainTransaction, bool) {
 	if orphanTransaction, ok := op.allOrphans[*transactionID]; ok {
+		if clone {
+			return orphanTransaction.Transaction().Clone(), true
+		}
 		return orphanTransaction.Transaction(), true
 	}
 	return nil, false
 }
 
-func (op *orphansPool) getAllOrphanTransactions() []*externalapi.DomainTransaction {
-	allOrphanTransactions := make([]*externalapi.DomainTransaction, 0, len(op.allOrphans))
+func (op *orphansPool) getOrphanTransactionsByAddresses(clone bool) (
+	sending model.ScriptPublicKeyStringToDomainTransaction,
+	receiving model.ScriptPublicKeyStringToDomainTransaction,
+	err error) {
+	sending = make(model.ScriptPublicKeyStringToDomainTransaction)
+	receiving = make(model.ScriptPublicKeyStringToDomainTransaction)
```

Comment:
we can have multiple outputs in one transaction, so `len(op.allOrphans)` would not suffice.

Reply:
> we can have multiple outputs in one transaction, so `len(op.allOrphans)` would not suffice.

It's only an initial allocation. It would save the need to grow the map multiple times.

Reply:
will do